gadgets/ci/image_inspect/test/unit/image_inspect_test.go

```diff
@@ -0,0 +1,260 @@
+		t.Fatalf("unmarshal ebpf.maps: %v", err)
+	}
+	sort.Slice(gotMaps, func(i, j int) bool {
+		return gotMaps[i].Name < gotMaps[j].Name
+	})
+	require.IsType(t, []extraInfoMap{}, gotMaps, "ebpf.maps: expected []extraInfoMap, got %T", gotMaps)
+	require.Equal(t, expectedMaps, gotMaps, "ebpf.maps mismatch")
+
+	// Validate eBPF variables
+	expectedVars := []string{"gadget_mapiter_qdisc___map1", "gadget_param_ifindex", "gadget_param_targ_ms", "ifindex", "targ_ms"}
+	execVarData := info.ExtraInfo.Data["ebpf.variables"].Content
+	var gotVars []extraInfoVariable
+	if err := json.Unmarshal(execVarData, &gotVars); err != nil {
+		t.Fatalf("unmarshal ebpf.variables: %v", err)
+	}
+	var gotVarNames []string
+	for _, v := range gotVars {
+		gotVarNames = append(gotVarNames, v.Name)
+	}
+	require.IsType(t, []extraInfoVariable{}, gotVars, "ebpf.variables: expected []extraInfoVariable, got %T", gotVars)
+	require.Len(t, gotVarNames, len(expectedVars), "ebpf.programs: expected %d programs, got %d", len(expectedVars), len(gotVarNames))
+	for _, prog := range expectedVars {
+		require.Contains(t, gotVarNames, prog, "ebpf.programs: %s not in got list", prog)
+	}
+
+	// Validate flowchart
+	expectedFlowchart := `flowchart LR
+map1[("map1")]
+map2[("map2")]
+consume_skb -- "Lookup+Delete" --> map2
+consume_skb -- "Lookup+Update" --> map1
+consume_skb["consume_skb"]
+kfree_skb -- "Lookup+Delete" --> map2
+kfree_skb -- "Lookup+Update" --> map1
+kfree_skb["kfree_skb"]
+qdisc_dequeue -- "Lookup+Delete" --> map2
+qdisc_dequeue -- "Lookup+Update" --> map1
+qdisc_dequeue["qdisc_dequeue"]
+qdisc_enqueue -- "Update" --> map2
+qdisc_enqueue["qdisc_enqueue"]
+`
+	require.IsType(t, ([]byte)(nil), info.ExtraInfo.Data["ebpf.flowchart"].Content, "ebpf.flowchart: expected string, got %T", info.ExtraInfo.Data["ebpf.flowchart"].Content)
+	gotFlowchart := string(info.ExtraInfo.Data["ebpf.flowchart"].Content)
+	require.Equal(t, expectedFlowchart, gotFlowchart, "flowchart mismatch")
+
+	// Validate sequence chart
+	expectedSequence := `sequenceDiagram
+box eBPF Programs
+participant consume_skb
+participant kfree_skb
+participant qdisc_dequeue
+participant qdisc_enqueue
+end
+box eBPF Maps
+participant map2
+participant map1
+end
+consume_skb->>map2: Lookup
+consume_skb->>map1: Lookup
+consume_skb->>map1: Update
+consume_skb->>map2: Delete
+kfree_skb->>map2: Lookup
+kfree_skb->>map1: Lookup
+kfree_skb->>map1: Update
+kfree_skb->>map2: Delete
+qdisc_dequeue->>map2: Lookup
+qdisc_dequeue->>map1: Lookup
+qdisc_dequeue->>map1: Update
+qdisc_dequeue->>map2: Delete
+qdisc_enqueue->>map2: Update
+`
+	require.IsType(t, ([]byte)(nil), info.ExtraInfo.Data["ebpf.sequence"].Content, "ebpf.sequence: expected string, got %T", info.ExtraInfo.Data["ebpf.sequence"].Content)
+	gotSequence := string(info.ExtraInfo.Data["ebpf.sequence"].Content)
+	require.Equal(t, expectedSequence, gotSequence, "sequence diagram mismatch")
+}
+
+func TestExtraInfoKeys(t *testing.T) {
```

Comment:
I think this test can be combined with the above one and then create helper function to validate ebpf/oci/wasm fields in the extraInfo. WDYT?

BTW, several oci fields can be tested, some other probably just validate they're not empty (e.g., digest, created). I think we could add a very simple metadata file to the image_inspect gadget so that we can validate it is included in the extraInfo.